### PR TITLE
Check entries value in partition load query to disallow negative values (#1351)

### LINKED ISSUE #1351: Query partition resource utilization should not allow "entries" parameter with negative values
Users can query partition resource utilization with the `entries` parameter. For example:

`GET http://<Cruise Control host>:<port>/kafkacruisecontrol/partition_load?entries=100`

If the `entries` parameter has negative values, for example `partition_load?entries=-123`, it would be treated as if it's 0. We should make it more strict by disallowing negative values and returning 4XX error code to indicate a bad request.

### PR DESCRIPTION
**Pull request by Lincong**
This PR resolves #1351

### CODE REVIEW

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/PartitionLoadParameters.java [hunk 1]
```diff
@@ -97,10 +97,14 @@ protected void initParameters() throws UnsupportedEncodingException {
     _endMs = ParameterUtils.endMs(_request);
     _partitionLowerBoundary = ParameterUtils.partitionBoundary(_request, false);
     _partitionUpperBoundary = ParameterUtils.partitionBoundary(_request, true);
-    _entries = ParameterUtils.entries(_request);
     _minValidPartitionRatio = ParameterUtils.minValidPartitionRatio(_request);
     _allowCapacityEstimation = ParameterUtils.allowCapacityEstimation(_request);
     _brokerIds = ParameterUtils.brokerIds(_request, true);
+    _entries = ParameterUtils.entries(_request);
```
Comment (efeg) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/PartitionLoadParameters.java:103]: Looks like `ParameterUtils #entries(HttpServletRequest)` is used by other callers. Would it ever be acceptable to get a negative value for entries? If not, should we move the sanity check to `ParameterUtils #entries(HttpServletRequest)`?

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/PartitionLoadParameters.java [hunk 2]
```diff
@@ -97,10 +97,14 @@ protected void initParameters() throws UnsupportedEncodingException {
...
+    if (_entries < 0) {
+      throw new UserRequestException("Invalid entries value. The entries value must be"
+          + " greater than or equal to 0. Got negative value: " + _entries);
```
Comment (efeg) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/PartitionLoadParameters.java:106]: Nit: Can we make this error message more concise? -- e.g. something like
```
throw new UserRequestException("The requested entries must be positive (Requested: " + _entries + ").");
```

**Review by efeg — COMMENTED** [review]
@Lincong thanks for the patch!
Left some comments.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/PartitionLoadParameters.java (hunk 1 above)
Comment (Lincong) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/PartitionLoadParameters.java:103]: That's a good idea. All `entries` should have positive values. I will move the check to `ParameterUtils`. Thanks

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/PartitionLoadParameters.java (hunk 2 above)
Comment (Lincong) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/PartitionLoadParameters.java]: Good idea!

**Review by efeg — APPROVED** [review]
LGTM!

**Pull request merged**